pkg/store/github/github_test.go

```diff
@@ -4,8 +4,192 @@
 
 package github
 
-import "testing"
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-github/v43/github"
+	"github.com/matryer/is"
+	"github.com/migueleliasweb/go-github-mock/src/mock"
+	"github.com/nrkno/terraform-registry/pkg/core"
+)
 
 func TestGithubStore(t *testing.T) {
-	t.Skip("not sure how to test this")
+	t.Run("create GitHubStore", func(t *testing.T) {
+		is := is.New(t)
+		emptyResult := new(github.RepositoriesSearchResult)
+		total := 0
+		emptyResult.Total = &total
+		mockedHTTPClient := mock.NewMockedHTTPClient(
+			mock.WithRequestMatch(
+				mock.GetSearchRepositories,
+				emptyResult,
+			),
+		)
+
+		c := github.NewClient(mockedHTTPClient)
+		store := &GitHubStore{
+			ownerFilter: "test-owner",
+			topicFilter: "test-topic",
+			client:      c,
+			cache:       make(map[string][]*core.ModuleVersion),
+		}
+
+		err := store.ReloadCache(context.Background())
+		is.True(err == nil)
+	})
+
+	t.Run("create GitHubStore with github error", func(t *testing.T) {
+		is := is.New(t)
+		mockedHTTPClient := mock.NewMockedHTTPClient(
+			mock.WithRequestMatchHandler(
+				mock.GetSearchRepositories,
+				http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					mock.WriteError(
+						w,
+						http.StatusInternalServerError,
+						"github went belly up or something",
+					)
+				}),
+			),
+		)
+		c := github.NewClient(mockedHTTPClient)
+		store := &GitHubStore{
+			ownerFilter: "test-owner",
+			topicFilter: "test-topic",
+			client:      c,
+			cache:       make(map[string][]*core.ModuleVersion),
+		}
+		store.client = c
+		err := store.ReloadCache(context.Background())
+		is.True(err != nil)
+		ghErr, ok := err.(*github.ErrorResponse)
+		if !ok {
+			t.Fatal("couldn't cast userErr to *github.ErrorResponse")
+		}
+
+		if ghErr.Message != "github went belly up or something" {
+			t.Errorf("user err is %s, want 'github went belly up or something'", err.Error())
+		}
+	})
+}
+
+func TestGetModuleVersion(t *testing.T) {
+	result := new(github.RepositoriesSearchResult)
+	total := 1
+	result.Total = &total
+	result.Repositories = []*github.Repository{
+		{
+			Name:     github.String("testrepo"),
+			FullName: github.String("test-owner/test-repo"),
+		},
+	}
+	mockedHTTPClient := mock.NewMockedHTTPClient(
+		mock.WithRequestMatch(
+			mock.GetSearchRepositories,
+			result,
+		),
+		mock.WithRequestMatch(
+			mock.GetReposTagsByOwnerByRepo,
+			[]github.RepositoryTag{
+				{
+					Name: github.String("v1.0.0"),
+				},
+			},
+		),
+	)
+
+	c := github.NewClient(mockedHTTPClient)
+	store := &GitHubStore{
+		ownerFilter: "test-owner",
+		topicFilter: "test-topic",
+		client:      c,
+		cache:       make(map[string][]*core.ModuleVersion),
+	}
+
+	err := store.ReloadCache(context.Background())
+	if err != nil {
```

Comment:
Comments on same line are printed to stderr along with the (failed) test results by `is`
```
is.NoErr(err) // Could not ReloadCache
```

Reply:
More of a FYI -- not important at all

Reply:
changed

Reply:
did not want to create an is instance on the outside, hence the if statement